crates/gitbutler-branch-actions/src/virtual.rs

```diff
@@ -508,6 +458,13 @@ pub fn list_virtual_branches_cached(
             requires_force = force // derive force requirement from the series
         }
 
+        let commits = series
+            .iter()
+            .cloned()
+            .filter_map(Result::ok)
+            .flat_map(|s| s.patches)
+            .collect();
+
```

Comment:
There is still a little bit of frontend logic, and a load of tests which rely on this list of commits, so we can't fully get rid of it. That said, pulling it out of the patches rather than calculating it twice in two slightly different ways is certainly an improvement